src/snowflake/cli/_plugins/nativeapp/release_directive/commands.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import logging
+from typing import Optional
+
+import typer
+from snowflake.cli._plugins.nativeapp.v2_conversions.compat import (
+    force_project_definition_v2,
+)
+from snowflake.cli._plugins.workspace.manager import WorkspaceManager
+from snowflake.cli.api.cli_global_context import get_cli_context
+from snowflake.cli.api.commands.decorators import with_project_definition
+from snowflake.cli.api.commands.flags import like_option
+from snowflake.cli.api.commands.snow_typer import SnowTyperFactory
+from snowflake.cli.api.entities.common import EntityActions
+from snowflake.cli.api.output.types import (
+    CollectionResult,
+    CommandResult,
+    MessageResult,
+)
+
+app = SnowTyperFactory(
+    name="release-directive",
+    help="Manages release directives of an application package",
+)
+
+log = logging.getLogger(__name__)
+
+
+@app.command("list", requires_connection=True)
+@with_project_definition()
+@force_project_definition_v2()
+def release_directive_list(
+    like: str = like_option(
+        help_example="`snow app release-directive list --like='my%'` lists all release directives starting with 'my'",
+    ),
+    channel: Optional[str] = typer.Option(
+        default=None,
+        show_default=False,
+        help="The release channel to use when listing release directives. If not provided, release directives from all release channels are listed.",
+    ),
```

Comment:
Should we hide the `channel` options until release channels are available in prod?

Reply:
They will be available in the upcoming release (in a week or 2), and they are optional anyways.
Snow CLI will not deploy before January, so I believe we should be ok.
Release channel commands  (snow app release-channel) will also follow so they will be visible regardless.